db/db_bloom_filter_test.cc

```diff
@@ -667,6 +670,400 @@ TEST_F(DBBloomFilterTest, BloomFilterReverseCompatibility) {
   }
 }
 
+/*
+ * A cache wrapper that tracks peaks and increments of filter
+ * construction cache reservation.
+ *        p0
+ *       / \   p1
+ *      /   \  /\
+ *     /     \/  \
+ *  a /       b   \
+ * peaks = {p0, p1}
+ * increments = {p1-a, p2-b}
+ */
+class FilterConstructResPeakTrackingCache : public CacheWrapper {
+ public:
+  explicit FilterConstructResPeakTrackingCache(std::shared_ptr<Cache> target)
+      : CacheWrapper(std::move(target)),
+        cur_cache_res_(0),
+        cache_res_peak_(0),
+        cache_res_increment_(0),
+        last_peak_tracked_(false),
+        cache_res_increments_sum_(0) {}
+
+  using Cache::Insert;
+  Status Insert(const Slice& key, void* value, size_t charge,
+                void (*deleter)(const Slice& key, void* value),
+                Handle** handle = nullptr,
+                Priority priority = Priority::LOW) override {
+    Status s = target_->Insert(key, value, charge, deleter, handle, priority);
+    auto map = CopyCacheDeleterRoleMap();
```

Comment:
Note: reading `CopyCacheDeleterRoleMap()` API leaves me with impression that I have to call this function every time a deleter is registered, which can be another overhead for this test.

Reply:
You can just compare to `GetNoopDeleterForRole<CacheEntryRole::kFilterConstruction>()` and for extra speed, save that ahead of time to a static variable or Test object field.